metrics.py

```diff
@@ -0,0 +1,65 @@
+"""Metrics different to sklearn are placed here. 
+Some of the functions are from CAML-MIMIC:
+(https://github.com/jamesmullenbach/caml-mimic/blob/master/evaluation.py)."""
+
+
+import numpy as np
+
+
+def intersect_size(y_true, y_pred, axis):
+    # axis=0 for label-level union (macro). axis=1 for instance-level
+    return np.logical_and(y_pred, y_true).sum(axis=axis).astype(float)
+
+
+def macro_precision(y_true, y_pred):
+    num = intersect_size(y_pred, y_true, 0) / (y_pred.sum(axis=0) + 1e-10)
+    return np.mean(num)
+
+
+def macro_recall(y_true, y_pred):
+    num = intersect_size(y_pred, y_true, 0) / (y_true.sum(axis=0) + 1e-10)
+    return np.mean(num)
+
+
+def macro_f1(y_true, y_pred):
+    prec = macro_precision(y_pred, y_true)
+    rec = macro_recall(y_pred, y_true)
+    if prec + rec == 0:
+        f1 = 0.
+    else:
+        f1 = 2*(prec*rec)/(prec+rec)
+    return f1
+
+
+def precision_at_k(y_true, y_pred, k):
+    # num true labels in top k predictions / k
+    sortd = np.argsort(y_pred)[:,::-1]
+    topk = sortd[:,:k]
+
+    # get precision at k for each example
+    vals = []
+    for i, tk in enumerate(topk):
+        if len(tk) > 0:
```

Comment:
Shouldn't len(tk) always be k?

Reply:
There could be multiple top-ks in CAML like `[8,15]`  for the full code, `[5]` for the top-50 code.
If we don't use multiple top-ks in the future, I should remove this.

Reply:
Wait, you're right!

Reply:
Remember to solve this!